internal/api/service/service.go

```diff
@@ -6,40 +6,47 @@ import (
 	"ship-it/internal/api/models"
 
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/helm/pkg/proto/hapi/release"
 )
 
-func New(l ReleaseLister) *Service {
-	return &Service{
-		lister: l,
-	}
+type ReleaseLister interface {
+	Get(ctx context.Context, namespace, release string) (*models.Release, error)
+	List(ctx context.Context, namespace string) ([]models.Release, error)
 }
 
-type ReleaseLister interface {
-	ListAll(ctx context.Context, namespace string) ([]models.Release, error)
+type ResourcesGetter interface {
+	Get(namespace, release string) (string, error)
+}
+
+func New(rl ReleaseLister, rg ResourcesGetter) *Service {
+	return &Service{
+		Namespace: v1.NamespaceAll,
+		releases:  rl,
+		resources: rg,
+	}
 }
 
 type Service struct {
-	lister ReleaseLister
+	Namespace string
+	releases  ReleaseLister
+	resources ResourcesGetter
 }
 
 func (s *Service) ListReleases(ctx context.Context) ([]models.Release, error) {
-	releases, err := s.lister.ListAll(ctx, v1.NamespaceAll)
-	if err != nil {
-		return nil, err
-	}
+	return s.releases.List(ctx, s.Namespace)
+}
 
-	for i := range releases {
-		r := &releases[i]
+func (s *Service) GetRelease(ctx context.Context, name string) (*models.Release, error) {
+	return s.releases.Get(ctx, s.Namespace, name)
+}
 
-		r.Status = s.getReleaseStatus(r).String()
+func (s *Service) GetReleaseResources(ctx context.Context, name string) (*models.ReleaseResources, error) {
+	resources, err := s.resources.Get(s.Namespace, name)
+	if err != nil {
+		return nil, err
 	}
 
-	return releases, err
-}
-
-func (s *Service) getReleaseStatus(_ *models.Release) release.Status_Code {
-	// The default state of a registered service is 'PENDING_INSTALL'.
-	// There's only one possible release status for now.
-	return release.Status_PENDING_INSTALL
```

Comment:
So we used to default any registered service to `PENDING_INSTALL`. The reasoning was that any newly registered service (ie. an associated `HelmRelease` exists in the cluster), would be installed by the operator as soon as it was noticed. However, this meant we couldn't tell if a release hadn't been noticed yet by the operator.

Now a service will be in a `UNKNOWN` state by default, until the operator becomes aware of it and begins to deploy it, at which point it'll update the `HelmReleaseStatus.Code` field to `PENDING_INSTALL`. So we can actually observe that initial state change.